docs/timer.md

```diff
@@ -0,0 +1,62 @@
+# Adding a Timer to a new Deployment
+
+There's documentation elsewhere about [how devices fit into the SwingSet
+architecture](devices.md). In order to install a Timer device, you first build
+a timer object in order to create the timer's endowments, source code, and 
+`poll()` function.
+
+```
+import { buildTimer, getTimerWrapperSourcePath } from `@agoric/swingset-vat`;
+const timer = buildTimer();
+```
+
+Then when configuring devices, the timer device can be added:
+
+```
+config.devices = [
+   ...
+  ['timer', timer.srcPath, timer.endowments],
+];
+```
+    
+and the Timer vat set up: 
+```
+config.vats.set('timer', { sourcepath: getTimerWrapperSourcePath() });
+```
+
+The `t.poll(now)` must be called periodically (usually at Block start time)
+with the current time. When `poll()` returns true, the kernel has work to do, 
+and we must call `controller.run()` and manage any resulting state updates.
+
+The "wrapper" vat has to provide a function to allow registering the device
+and creating the service. The vat is responsible for creating repeater objects
+to wrap the device node as a capability that can be accessed by regular vat
+code.
+
+In `bootstrap()` for a particular SwingSet, we create a timerService, and
+make it accessible to the user in `home`
+
+```
+await vats.timerWrapper~.registerTimerDevice(devices.timer);
+const timerService = vats.timerWrapper~.createTimerService();
+```
+
+Then users in the REPL can use the timerService to schedule wakeups.
+
+```
+const timestamp = timerService~.getCurrentTimestamp();
```

Comment:
do these examples need `await`? Or maybe the variable should be named `timestampP` or something to make it clear that this is a promise.

Reply:
hm, I guess I'm imagining someone copying this code into their vat, in which case 'timestamp' isn't really a timestamp, it's a promise for one, and the difference should be clear in the docs (e.g. you couldn't do `timestamp + 30*60` to display the expected expiration time of a 30-minute contract). But we'll probably get more feedback from users, and can refine it later.